### PR TITLE
Replaced WebRequest with RestSharp

### PR DESCRIPTION
**Pull request by eran-gil**
Removed support for .net 3.5.
Would like someone to test it with aws.
Also haven't tested what happens when using it with .net core.

### CODE REVIEW

**Comment by eran-gil** [comment]
Updated the dependency for .net core and it's working :)

**Comment by eran-gil** [comment]
This answers issue #47

File: log4stash.nuspec [hunk 1]
```diff
@@ -36,17 +36,19 @@
...
-		</group>
+		
 		<group targetFramework=".NETFramework4.0">
 			<dependency id="log4net" version="2.0.8.0" />
+			<dependency id="RestSharp" version="105.2.3" />
+			<dependency id="Microsoft.Bcl.Async" version="1.0.168" />
 		</group>
 		<group targetFramework=".NETFramework4.5">
 			<dependency id="log4net" version="2.0.8.0" />
+			<dependency id="RestSharp" version="105.2.3" />
```
Comment (urielha) [inline log4stash.nuspec:47]: Microsoft.Bcl.Async ?

File: src/log4stash/ElasticClient/IElasticsearchClient.cs [hunk 2]
```diff
@@ -13,6 +13,6 @@ public interface IElasticsearchClient : IDisposable
         AuthenticationMethodChooser AuthenticationMethod { get; set; }
         void PutTemplateRaw(string templateName, string rawBody);
         void IndexBulk(IEnumerable<InnerBulkOperation> bulk);
-        IAsyncResult IndexBulkAsync(IEnumerable<InnerBulkOperation> bulk);
+        bool IndexBulkAsync(IEnumerable<InnerBulkOperation> bulk);
```
Comment (urielha) [inline src/log4stash/ElasticClient/IElasticsearchClient.cs]: I think no one use this `bool` result and the `IAsyncResult` has been there by mistake (or someone remove the reference without changing to void).
Anyway, I suppose you can change it to `void` 🤷‍♂️

File: log4stash.nuspec [hunk 3]
```diff
@@ -36,17 +36,19 @@
...
 		</group>
 		<group targetFramework=".NETStandard2.0">
 			<dependency id="log4net" version="2.0.8.0" />
+			<dependency id="RestSharp" version="106.1.0" />
```
Comment (urielha) [inline log4stash.nuspec:51]: Here too - Microsoft.Bcl.Async ? or it doesn't needed?

File: src/log4stash/Authentication/AwsAuthenticationMethod.cs [hunk 4]
```diff
@@ -15,20 +18,26 @@ public class AwsAuthenticationMethod : IAuthenticationMethod
 
         public string CreateAuthenticationHeader(RequestData requestData)
```
Comment (urielha) [inline src/log4stash/Authentication/AwsAuthenticationMethod.cs]: Can we delete `CreateAuthenticationHeader` ? I didn't find any reference to it :)

**Comment by urielha** [comment]
Wow! thanks a lot @erangil2 😃 
I added just few comments

File: log4stash.nuspec (hunk 1 above)
Comment (eran-gil) [reply log4stash.nuspec:47]: It's required to support proper async in .net 4.

File: log4stash.nuspec (hunk 3 above)
Comment (eran-gil) [reply log4stash.nuspec:51]: It's required to support proper async in .net 4.

**Pull request merged**